volmdlr/faces.py

```diff
@@ -8242,5 +8261,21 @@ def intersection(self, shell2, tol=1e-8):
         faces = self.intersection_faces(shell2, intersecting_faces, intersecting_combinations)
         faces += self.get_non_intersecting_faces(shell2, intersecting_faces, intersection_method=True) +\
             shell2.get_non_intersecting_faces(self, intersecting_faces, intersection_method=True)
+        faces = self.validate_bool_operations_closedshell_faces(faces)
         new_shell = ClosedShell3D(faces)
         return [new_shell]
+
+    def validate_bool_operations_closedshell_faces(self, new_faces):
+        faces_graph = nx.Graph()
```

Comment:
Could it be an attribute of shells this graph?

Reply:
yes!! I will add it then

Reply:
Done